packages/smithy-client/src/parse-utils.ts

```diff
@@ -65,3 +65,27 @@ export function expectString(value: any): string | undefined {
   }
   throw new TypeError(`Expected string, got ${typeof value}`);
 }
+
+/**
+ * Asserts a value is a number and returns it, and also converts string
+ * representations of non-numeric floats into Numbers.
+ *
+ * @param value A number or string representation of a non-numeric float.
+ * @returns The value as a number, undefined if it's null/undefined,
+ *     otherwise an error is thrown.
+ */
+export const handleFloat = (value: string | number): number | undefined => {
```

Comment:
`parseFloat`, like `parseBoolean` above?

Reply:
`parseFloat` is a built in function